src/Tickets/Commerce/Gateways/Square/REST/Webhook_Endpoint.php

```diff
@@ -0,0 +1,378 @@
+<?php
+
+/**
+ * Square Webhook Endpoint
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Commerce\Gateways\Square\REST
+ */
+
+namespace TEC\Tickets\Commerce\Gateways\Square\REST;
+
+use TEC\Tickets\Commerce\Gateways\Contracts\Abstract_REST_Endpoint;
+use TEC\Tickets\Commerce\Gateways\Square\Webhooks;
+use TEC\Tickets\Commerce\Gateways\Square\Order;
+use WP_REST_Request;
+use WP_REST_Server;
+use WP_REST_Response;
+use WP_Error;
+
+/**
+ * Class Webhook_Endpoint.
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Commerce\Gateways\Square\REST
+ */
+class Webhook_Endpoint extends Abstract_REST_Endpoint {
+
+	/**
+	 * The REST namespace for this endpoint.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected string $namespace = 'tribe/tickets/v1';
+
+	/**
+	 * The REST endpoint path for this endpoint.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected string $path = '/commerce/square/webhooks';
+
+	/**
+	 * Get the namespace for this endpoint.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_namespace(): string {
+		return $this->namespace;
+	}
+
+	/**
+	 * Get the path for this endpoint.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_path(): string {
+		return $this->path;
+	}
+
+	/**
+	 * Checks if the current user has permissions to the endpoint.
+	 * For webhooks, we skip permission checks because this is called by Square,
+	 * we validate the request using the webhook signature instead.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return bool Always returns true as we validate using the webhook signature.
+	 */
+	public function has_permission( WP_REST_Request $request ) {
+		return true;
+	}
```

Comment:
```suggestion
```

Reply:
I will add permissions later. This is a placehodler for now.